source/blender/gpu/intern/gpu_material.c

```diff
@@ -877,18 +882,85 @@ static void shade_light_textures(GPUMaterial *mat, GPULamp *lamp, GPUNodeLink **
 		MTex *mtex = lamp->la->mtex[i];
 
 		if (mtex && mtex->tex->type & TEX_IMAGE && mtex->tex->ima) {
-			mat->dynproperty |= DYN_LAMP_PERSMAT;
+			if (lamp->type == LA_AREA) {
+				mat->dynproperty = DYN_LAMP_VEC | DYN_LAMP_CO | DYN_LAMP_AREAMAT;
+			}
+			else {
+				mat->dynproperty |= DYN_LAMP_PERSMAT;
+			}
 
 			float one = 1.0f;
 			GPUNodeLink *tex_rgb;
 
 			GPU_link(mat, "shade_light_texture",
-			         GPU_builtin(GPU_VIEW_POSITION),
-			         GPU_image(mtex->tex->ima, &mtex->tex->iuser, false), GPU_uniform(mtex->size),
+					 GPU_builtin(GPU_VIEW_POSITION),
+					 GPU_image(mtex->tex->ima, &mtex->tex->iuser, false), GPU_uniform(mtex->size),
 					 GPU_select_uniform(&mtex->lodbias, GPU_DYNAMIC_TEX_LODBIAS, NULL, mat->ma),
-			         GPU_dynamic_uniform((float *)lamp->dynpersmat, GPU_DYNAMIC_LAMP_DYNPERSMAT, lamp->ob),
-			         &tex_rgb);
-			texture_rgb_blend(mat, tex_rgb, *rgb, GPU_uniform(&one), GPU_uniform(&mtex->colfac), mtex->blendtype, rgb);
+					 GPU_dynamic_uniform((float *)lamp->dynpersmat, GPU_DYNAMIC_LAMP_DYNPERSMAT, lamp->ob), &tex_rgb);
+
+			if (lamp->type == LA_AREA) {
+				one = 0.95f;
+				texture_rgb_blend(mat, tex_rgb, *rgb, GPU_uniform(&one), GPU_uniform(&mtex->colfac), MTEX_MUL, rgb); 
+			}
+			else {
+				texture_rgb_blend(mat, tex_rgb, *rgb, GPU_uniform(&one), GPU_uniform(&mtex->colfac), mtex->blendtype, rgb);
+			}
+		}
+	}
+}
+
+static void shade_area_diff_texture(GPUMaterial *mat, GPULamp *lamp, GPUNodeLink **rgb, GPUShadeInput *shi)
```

Comment:
I don't know, the patch is just like that. And I can't contact the original owners